pkgs/applications/graphics/gimp/default.nix

```diff
@@ -135,84 +173,144 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
-  preConfigure = ''
-    # The check runs before glib-networking is registered
-    export GIO_EXTRA_MODULES="${glib-networking}/lib/gio/modules:$GIO_EXTRA_MODULES"
+  postPatch = ''
+    patchShebangs \
+      app/tests/create_test_env.sh \
+      tools/gimp-mkenums
+
+    # GIMP is executed at build time so we need to fix this.
+    # TODO: Look into if we can fix the interp thing.
+    chmod +x plug-ins/python/{colorxhtml,file-openraster,foggify,gradients-save-as-css,histogram-export,palette-offset,palette-sort,palette-to-gradient,python-eval,spyro-plus}.py
+    patchShebangs \
+      plug-ins/python/{colorxhtml,file-openraster,foggify,gradients-save-as-css,histogram-export,palette-offset,palette-sort,palette-to-gradient,python-eval,spyro-plus}.py
+  '';
+
+  preBuild = ''
+    # Our gobject-introspection patches make the shared library paths absolute
+    # in the GIR files. When running GIMP in build or check phase, it will try
+    # to use plug-ins, which import GIMP introspection files which will try
+    # to load the GIMP libraries which will not be installed yet.
+    # So we need to replace the absolute path with a local one.
+    # We are using a symlink that will be overridden during installation.
+    mkdir -p "$out/lib"
+    ln -s "$PWD/libgimp/libgimp-3.0.so.0" "$out/lib/libgimp-3.0.so.0"
+    ln -s "$PWD/libgimpbase/libgimpbase-3.0.so.0" "$out/lib/libgimpbase-3.0.so.0"
+    ln -s "$PWD/libgimpcolor/libgimpcolor-3.0.so.0" "$out/lib/libgimpcolor-3.0.so.0"
+    ln -s "$PWD/libgimpconfig/libgimpconfig-3.0.so.0" "$out/lib/libgimpconfig-3.0.so.0"
+    ln -s "$PWD/libgimpmath/libgimpmath-3.0.so.0" "$out/lib/libgimpmath-3.0.so.0"
+    ln -s "$PWD/libgimpmodule/libgimpmodule-3.0.so.0" "$out/lib/libgimpmodule-3.0.so.0"
+  '';
+
+  preCheck = ''
+    # Avoid “Error retrieving accessibility bus address”
+    export NO_AT_BRIDGE=1
+    # Fix storing recent file list in tests
+    export HOME="$TMPDIR"
+    export XDG_DATA_DIRS="${glib.getSchemaDataDirPath gtk3}:${adwaita-icon-theme}/share:$XDG_DATA_DIRS"
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    meson test --timeout-multiplier 4 --print-errorlogs
+
+    runHook postCheck
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(--prefix PATH : "${
+      lib.makeBinPath [
+        # for dot for gegl:introspect (Debug » Show Image Graph, hidden by default on stable release)
+        graphviz
+      ]
+    }:$out/bin")
   '';
 
   postFixup = ''
-    wrapProgram $out/bin/gimp-${lib.versions.majorMinor finalAttrs.version} \
-      --set GDK_PIXBUF_MODULE_FILE "$GDK_PIXBUF_MODULE_FILE"
+    # Cannot be in postInstall, otherwise _multioutDocs hook in preFixup will move right back.
+    moveToOutput "share/doc" "$devdoc"
   '';
 
   passthru = {
     # The declarations for `gimp-with-plugins` wrapper,
     # used for determining plug-in installation paths
-    majorVersion = "${lib.versions.major finalAttrs.version}.0";
+    majorVersion = "3.0";
```

Comment:
Is this correct, that this isn't following semver?

Reply:
This was hardcoded for development releases which used 2.99.x for the version number but already had 3.0 for ABI version.

Reply:
```suggestion
    majorVersion = "${lib.versions.major finalAttrs.version}.0";
```